src/main/scala/is/hail/io/LoadMatrix.scala

```diff
@@ -23,160 +23,47 @@ object LoadMatrix {
   }
 
   // this assumes that col IDs are in last line of header.
-  def parseHeader(line: String, sep: String = "\t", hasRowIDName: Boolean = false): Array[String] =
-    if (hasRowIDName)
-      line.split(sep).tail
-    else
-      line.split(sep)
+  def parseHeader(line: String, sep: String = "\t", nAnnotations: Int, annotationHeaders: Option[Seq[String]]): (Array[String], Array[String]) =
+    annotationHeaders match {
+      case None =>
+        val r = line.split(sep)
+        (r.slice(0, nAnnotations), r.slice(nAnnotations, r.length))
```

Comment:
What if the length is less than nAnnotations?  Probably should generate a message.